Modules/_sqlite/cursor.c

```diff
@@ -925,6 +931,11 @@ _pysqlite_query_execute(pysqlite_Cursor* self, int multiple, PyObject* operation
         if (!parameters) {
             break;
         }
+        // PyIter_Next() may have a side-effect on the connection's state.
```

Comment:
Note: if `parameters` is NULL and we have an active exception, we break the loop. It should only happen when we are having a custom iterator, that is, `multiple = 1`. As such, we wouldn't branch into the `if (!multiple)` and we wouldn't call `PyLong_FromLongLong`.

However, I'll add tomorrow some assertions because it's not entirely clear from the code alone.